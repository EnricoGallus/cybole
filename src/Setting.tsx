import {useEffect} from "react";

function Setting() {
    useEffect(() => {
        const storedPreference = localStorage.getItem('darkModePreference');
    })

    return (
        <div>
            Setting
        </div>
    )
}

export default Setting;