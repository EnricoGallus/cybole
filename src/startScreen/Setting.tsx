import React, {useEffect, useState} from 'react';
import {Button} from "primereact/button";
import {InputText} from "primereact/inputtext";
import {open} from "@tauri-apps/api/dialog";
import persister from "../utils/persister";

const Setting = () => {
    const [pathToCybop, setPathToCybop] = useState('');

    useEffect(() => {
        persister.get().get('pathToCybop').then(value => setPathToCybop(value as string));
    }, [])

    const selectPathToCybop = async () => {
        const selected = await open({
            directory: false,
            multiple: false,
        });

        if (selected !== null) {
            setPathToCybop(selected as string)
        }
    };

    const saveSettings = () => {
        persister.get().set('pathToCybop', pathToCybop);
    }

    return (
        <div className="card grid">
            <div className="col-12">
                <h5>Settings</h5>
            </div>
            <div className="col-12">
                <div className="field">
                    <label htmlFor="pathToCybop" className="block">Path to Cybop</label>
                    <InputText id="pathToCybop" className="block" value={pathToCybop as string} onClick={selectPathToCybop} />
                </div>
            </div>
            <div className="col-12">
                <Button id="saveSettings" onClick={saveSettings}>Save</Button>
            </div>
        </div>);
}

export default Setting;
