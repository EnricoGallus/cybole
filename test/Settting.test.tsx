import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Setting from '../src/startScreen/Setting';

describe('Setting', () => {
    test('renders component', () => {
        const { container } = render(<Setting />, {
            wrapper: MemoryRouter,
        });

        expect(container).toMatchSnapshot();
    });
});
