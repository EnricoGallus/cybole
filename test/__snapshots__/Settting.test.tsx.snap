// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting renders component 1`] = `
<div>
  <div
    class="card grid"
  >
    <div
      class="col-12"
    >
      <h5>
        Settings
      </h5>
    </div>
    <div
      class="col-12"
    >
      <div
        class="field"
      >
        <label
          class="block"
          for="pathToCybop"
        >
          Path to Cybop
        </label>
        <input
          class="p-inputtext p-component block"
          id="pathToCybop"
          value=""
        />
      </div>
    </div>
    <div
      class="col-12"
    >
      <button
        class="p-button p-component"
        id="saveSettings"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
